### PR TITLE
Fix iOS app configuration missing certain specifications

### LINKED ISSUE #7443: iOS app submission guideline regression
As of the latest build, we regressed what was previously fixed (but have fixed importing files in the process).

We will need to address this as it will mean the app cannot be submitted in the future.

> ITMS-90737: Missing Document Configuration - By declaring the CFBundleDocumentTypes key in your app, you've indicated that your app is able to open documents. Please set the UISupportsDocumentBrowser key to "YES" if your app uses a UIDocumentBrowserViewController. Otherwise, set the LSSupportsOpeningDocumentsInPlace key in the Info.plist to "YES" (recommended) or "NO" to specify whether the app can open files in place. All document-based apps must include one of these configurations. For more information, visit https://developer.apple.com/document-based-apps/.

### PR DESCRIPTION
**Pull request by frenzibyte**
- Closes https://github.com/ppy/osu/issues/7443

After having this one simmer for some time while working on other tasks that require UIKit's document-related components, the specification in this PR here *should* be correct for what osu! is as an iOS app.

osu! is *not* a document-based app, contrary to https://github.com/ppy/osu/issues/7443#issuecomment-571576554. None of its files should be accessible to the user, not even logs, that's the last kind of data I would expose to a user that wants to play a game on their phone. The only kind of files that can be exposed to the user is exported beatmaps/skins/scores, and even then that doesn't justify it to be a document-based app at all.

osu! should not support opening documents in place either, there is nothing in the game that requires opening documents in the user's storage and write to it.

In conclusion, to address `ITMS-90737`:
 - `UISupportsDocumentBrowser` should not be set to YES as osu! is not a document-based app
 - `LSSupportsOpeningDocumentsInPlace` should be set to NO as osu! does not open documents in place.

Tested the iOS file selector/presenter implementations to still work with this key set.

### CODE REVIEW

**Comment by smoogipoo** [comment]
These days we also get:

```
ITMS-90683: Missing purpose string in Info.plist - Your app’s code references one or more APIs that access sensitive user data, or the app has one or more entitlements that permit such access. The Info.plist file for the “[osu.iOS.app](http://osu.ios.app/)” bundle should contain a NSBluetoothAlwaysUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. If you’re using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. For details, visit: https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy/requesting_access_to_protected_resources.
```

Probably worth fixing here too, if possible.

**Comment by frenzibyte** [comment]
It's weird that we require bluetooth permissions at all, and I'm not sure what kind of message would satisify this warning and not have the game being rejected so I investigated it a little. Turns out it was SDL referencing CoreBluetooth via an API it provides for HID support (named HIDAPI). To avoid unnecessary hassles, I've took the time to disable CoreBluetooth from SDL instead, as in ppy/SDL3-CS#188.

This also involved removing iOS from ppy/SDL2-CS completely, as SDL2 also references CoreBluetooth and I have no idea how to disable it there (I tried doing what I did in SDL3 but no go). PR: https://github.com/ppy/SDL2-CS/pull/197

**Review by smoogipoo — DISMISSED** [review]

**Comment by smoogipoo** [comment]
Do we lose anything by excluding hidapi? For example external keyboard support, apple pen support?

If so, maybe best to just state that in the requirement rather than disabling?

**Comment by frenzibyte** [comment]
I don't think SDL's HIDAPI stuff has anything to do with functionalities already provided by Apple's native frameworks itself (hardware keyboard via `GCKeyboard` and apple pen via multiple UIKit APIs), but I admit I haven't tested this to be 101% sure so I'll do to confirm.

**Comment by smoogipoo** [comment]
I believe they're doing their own stuff. That said I think the simplest path might just be to say we don't use these resources? In-case someone _does_ use them and wonders why we don't include them in our SDL-CS releases.

> While your app might not use these APIs, a purpose string is still required.

It's a bit wishy-washy but seems to indicate "we don't actually use this" is somewhat of an appropriate response.

**Comment by peppy** [comment]
As long as it doesn't end up in a user facing popup. These strings generally do, in my understanding, when they are actually accessed.

That said, trimming our SDL releases down to only what we use seems like a good direction regardless?

**Comment by smoogipoo** [comment]
AFAICR our testflight build doesn't have any popups. If we can get away with not trimming and also not have a popup show, then that's a win-win in my book, given that these projects have external consumers. https://github.com/ppy/SDL2-CS/network/dependents https://github.com/ppy/SDL3-CS/network/dependents

**Comment by smoogipoo** [comment]
Here's a similar issue for reference: https://stackoverflow.com/a/58009150

**Comment by frenzibyte** [comment]
Okay, my main intention was to find which API was using `CoreBluetooth` so as to make sure we're not hitting a code path that uses it. Now that we're aware where it's being used and it's something that we would *never* hit, I'll go ahead with a "we don't use this" string for now and hope it doesn't mess up the app store review process.

**Comment by smoogipoo** [comment]
Shall we close SDL-CS PRs then?

**Comment by frenzibyte** [comment]
Yep.
